website/src/content/development/adding-typescript.mdx

```diff
@@ -0,0 +1,95 @@
+```
+
+The TypeScript starter template is the same as the standard JS starter template in terms of functionality but it includes the additional TypeScript setup.
+
+# Configuration
+
+Every TypeScript project requires a `tsconfig.json` file to instruct TypeScript on how to compile the project.
+
+Therefore, we provide a base TSConfig to be used in your `tsconfig.json` file:
+
+```json
+{
+  "extends": "@commercetools-frontend/application-config/tsconfig-mc-app.json"
+}
+```
+
+Furthermore, we provide a `client.d.ts` declaration file with some basic type shims for importing media assets:
+
+- `.mod.css` and `.module.css`
+- `.png`
+- `.svg`
+
+You can include this using the TypeScript triple-slash directives:
+
+```ts
+/// <reference types="@commercetools-frontend/application-config/client" />
+```
+
+<Info>
+
+By default, this is included in the TypeScript starter template `src/index.tsx` entry point file.
+
+</Info>
+
+You can also include this in the `tsconfig.json` file in the `compilerOptions.types` field but we don't recommend
+to use that unless you are very familiar with the [implications of using the `types` field](https://www.typescriptlang.org/tsconfig#types).
+
+Additional adjustments to other config files is also required, in particular:
+
+- `.prettierrc`: use the `typescript` parser.
+- `jest.test.config.js`: use the `@commercetools-frontend/jest-preset-mc-app/typescript` preset.
+- `jest.eslint.config.js`: include the file extensions `ts` and `tsx` in `moduleFileExtensions`, then `<rootDir>/**/*.ts` and `<rootDir>/**/*.tsx` in `testMatch`.
+
+# Migrate to TypeScript
+
+If you have an existing Custom Application project and would like to migrate it to TypeScript, you need to do the following:
+
+* Install the `typescript` dependency.
+* Add a `tsconfig.json` file. See [configuration](#configuration) for the tooling setup.
+* Migrate the JavaScript files to TypeScript, either `.ts` or `.tsx` (if a file contains JSX syntax).
+
+  <Info>
+
+  Migrating source files to TypeScript can be done incrementally. We recommend starting from the "leaf files" and work your way up to the application entry point.
+  The [migrating from JavaScript documentation](https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html) can help with setting up a basic plan.
+
+  All UI Kit and ApplicationKit packages provide declaration files and export useful types when necessary.
+
+  During the migration you might need to loosen up the types strictness, for example by allowing explicit `any` types.
+
+  ```json title="tsconfig.json"
+  {
+    "extends": "@commercetools-frontend/application-config/tsconfig-mc-app.json",
+    "compilerOptions": {
+      "noExplicitAny": false,
+      "strict": false
+    }
+  }
+  ```
+
+  Please remember to revert that once the migration is over, as using `any` should be avoided.
+
+  </Info>
+
+* Add a script command `"typecheck": "tsc --noEmit"` in your `package.json` to compile the application.
```

Comment:
For me this is a little confusing since the command is called `typecheck`, which I understand as a command to validate the types in the application but we say this is to compile your application.

I thought our `mc-scripts build` command would also support building the TS application and the new `typecheck` command is only for "lintin" purposes.

Reply:
Good point. Technically "compiling" is what TypeScript does. In fact, running `tsc` does compile the project but the option `--noEmit` does not emit any files, which makes it look like it only checks the types.

To be honest this is confusing yes, a more explicit command would have been better, something like `tsc check` or something.

Let's see if we can try rewording this:

```suggestion
* Add a script command `"typecheck": "tsc --noEmit"` in your `package.json` to run the type checks of the application.
```